app/scripts/background.js

```diff
@@ -496,6 +503,47 @@ function emitDappViewedMetricEvent(
   }
 }
 
+/**
+ * Track dapp connection when loaded and permissioned
+ *
+ * @param {Port} remotePort - The port provided by a new context.
+ * @param {object} preferencesController - Preference Controller to get total created accounts
+ * @param {object} permissionController - Permission Controller to check if origin is permitted
+ */
+function trackDappView(
+  remotePort,
+  preferencesController,
```

Comment:
Nit: We've been trying to move away from passing around controller references as arguments, as we found it makes it easy to miss things when making breaking changes to controllers.

I see that this is pre-existing in this case though, so maybe we can clean this up later.

Reply:
is the preference to pass a callback in instead?

Reply:
Exactly. The preferences is to use the messenger, or callbacks for code not setup with a messenger.

Reply:
using the messenger where possible now https://github.com/MetaMask/metamask-extension/pull/25075/commits/ab7767b911e5f74684cb519e6841cb70e73b9e6e